test/index.ts

```diff
@@ -0,0 +1,115 @@
+import tape from "fresh-tape";
+
+import { Member } from "../src/member";
+
+const EXAMPLE_ID = "example";
+
+tape("Able to initialize a member", (t) => {
+  const member = new Member();
+
+  member.processFeeds();
+
+  t.pass("Able to process feeds with zero data");
+
+  t.end();
+});
+
+tape("Able to add a member by ID", (t) => {
+  const member = new Member();
+
+  const req = member.requestAdd(EXAMPLE_ID);
+
+  t.ok(req, "Generated a request");
+
+  const res = member.acceptRequest(req);
+
+  t.ok(res, "Generated response");
+
+  member.processFeeds();
+
+  t.pass("Processed feeds");
+
+  const expectedMembers = [member.id, EXAMPLE_ID];
+  t.deepEqual(
+    member.knownMembers,
+    expectedMembers,
+    "New member got added to list"
+  );
+
+  t.end();
+});
+
+tape("Able to add a member by ID and sync", (t) => {
+  const member = new Member();
+  const other = new Member({ initiator: member.id });
+
+  const req = member.requestAdd(other.id);
+
+  t.ok(req, "Generated a request");
+
+  const res = member.acceptRequest(req);
+
+  t.ok(res, "Generated response");
+
+  member.processFeeds();
+
+  other.sync(member);
+
+  t.pass("Able to sync with member");
+
+  other.processFeeds();
+
+  t.pass("Able to process feeds after sync");
+
+  const expectedMembers = [member.id, other.id];
+  t.deepEqual(
+    other.knownMembers,
+    expectedMembers,
+    "New member saw itself added to list"
+  );
+
+  t.end();
+});
+
+tape("Process request by syncing one peer at a time", (t) => {
+  const members = initializeMembers(5);
+
+  let previous = members[0];
+  previous.requestAdd(EXAMPLE_ID);
+  previous.processFeeds();
+
+  for (let next of members.slice(1)) {
+    sync(previous, next);
+    previous = next;
+  }
+
+  const wasAdded = previous.knownMembers.includes(EXAMPLE_ID);
+  t.ok(wasAdded, "Request got processed successfully");
+
+  t.end();
+});
+
```

Comment:
Essential tests here are the remove/add processes.

Reply:
Adding TODOs to the README to keep track of these, will work on them this week